### PR TITLE
PR: DVCI-144 part 1: Error handling

### PR DESCRIPTION
**Pull request by pmadria**
Now error handling is working with a patient not found or multiple matches found depending on what the situation is with the appropriate codes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,10 +4,13 @@
   "dependencies": {
     "@babel/preset-react": "^7.14.5",
     "@date-io/date-fns": "1.x",
+    "@emotion/react": "^11.4.1",
+    "@emotion/styled": "^11.3.0",
     "@fontsource/roboto": "^4.5.0",
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.11.2",
     "@material-ui/pickers": "^3.3.10",
+    "@mui/material": "^5.0.0",
```
Comment (radamson) [inline package.json:13]: I think this relates to my above comment, but this is adding material-ui v5 alongside v4. I'm all for updating our dependencies here, but I don't like having two versions of material-ui sticking around without a plan to update.

Can we just use the v4 components for now (https://v4.mui.com/components/alert/#alert) and then have a separate PR to update the material-ui dependencies?

File: package.json [hunk 2]
```diff
@@ -4,10 +4,13 @@
   "dependencies": {
     "@babel/preset-react": "^7.14.5",
     "@date-io/date-fns": "1.x",
+    "@emotion/react": "^11.4.1",
+    "@emotion/styled": "^11.3.0",
```
[the remark this answers is not in the document]
Comment (pmadria) [reply package.json:8]: Yep I added this originally but then changed methods of doing it and forgot to remove it from the package json

Resolved it

File: package.json (hunk 1 above)
Comment (pmadria) [reply package.json]: Yeah that's a good point - can lead to problems. I removed the v5 version and replaced with v4.

Resolved

**Review by ssayer — APPROVED** [review]
I'll defer for @radamson on the actual message content, but works as expected for me.

File: client/src/components/Form/Form.jsx [hunk 3]
```diff
@@ -31,15 +36,30 @@ const Form = () => {
...
-      .then((response) => response.json())
+      .then((response) => {
+        if (response.ok){
+          response.json()
+        }
+        else if (response.status == 404){
+          throw Error('Patient not found');
+        } 
+        else if (response.status == 422){
+          throw Error ('Multiple Matches. Please enter more information.');
```
Comment (mickohanlon23) [inline client/src/components/Form/Form.jsx:47]: We might want to align with the [IIS Issuer Reference Designs for the text here](https://projects.invisionapp.com/share/5M11EMPTJH7N#/screens/455685499?browse). There is text there for the multi-match error. I don't see any specific instruction for Patient Not Found, but what you have for that error message seems clear to me.

File: package.json [hunk 4]
```diff
@@ -7,6 +7,7 @@
     "@fontsource/roboto": "^4.5.0",
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.11.2",
+    "@material-ui/lab": "^4.0.0-alpha.60",
```
Comment (mickohanlon23) [inline package.json:10]: Do we also want to commit the new addition to `yarn.lock` after adding `@material-ui/lab`? (Mine updated after running `yarn install`)

File: client/src/components/Form/Form.jsx [hunk 5]
```diff
@@ -4,13 +4,18 @@ import useStyles from './styles';
...
+import Alert from '@material-ui/lab/Alert';
+
 
 const Form = () => {
   const styles = useStyles();
   const history = useHistory();
   const [firstName, setFirstName] = useState(null);
   const [lastName, setLastName] = useState(null);
   const [selectedDate, handleDateChange] = useState(null);
+  const [phoneNumber, setPhoneNumber] = useState(null);
```
Comment (mickohanlon23) [inline client/src/components/Form/Form.jsx:16]: Why is `phoneNumber` being added if it's not being used? Should we hold off on setting this state variable until it actually has functionality?

Comment (pmadria) [reply client/src/components/Form/Form.jsx:16]: I forgot to remove that was also working on the other fields. Good catch.

Removed it.

File: client/src/components/Form/Form.jsx (hunk 3 above)
Comment (pmadria) [reply client/src/components/Form/Form.jsx:47]: Ah didn't see that - good catch. Aligned the text and slight modification to Patient Not Found

File: package.json (hunk 4 above)
Comment (pmadria) [reply package.json:10]: I can commit the package-lock as well

Comment (mickohanlon23) [reply package.json:10]: We shouldn't have both. I would only update yarn.lock, and remove the package-lock file since we're using yarn for this project.

**Review by mickohanlon23 — APPROVED** [review]
Both error messages appear to be formatted correctly and are triggered when expected.

**Pull request merged**